v8.3.0 (#17740)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.3.0

> _May 7, 2025_

Reviewer: ```suggestion
_May 8, 2025_
```

We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:

> TODO INSERT HIGHLIGHTS

Reviewer: Any Charts changes worth highlighting?
```suggestion
- 🐞 Bugfixes
```

> - 🐞 Bugfixes

Reviewer 2: ```suggestion
- 📚 Documentation improvements
- 🐞 Bugfixes
```

Special thanks go out to this community member for a valuable contribution:
@ptuukkan.
Team members who have contributed to this release:
@alexfauquette, @arminmeh, @bernardobelchior, @flaviendelangle, @Janpot, @JCQuintas, @LukasTy, @MBilalShafi, @rita-codes, @romgrk.

### Data Grid

#### `@mui/x-data-grid@8.3.0`

- [DataGrid] Fix cell editing of computed columns with data source (#17684) @ptuukkan
- [DataGrid] Fix lazy loading crash with `isRowSelectable` prop (#17629) @MBilalShafi
- [DataGrid] Fix: use CSS nonce (#17726) @romgrk

> - [DataGrid] Ignore `preProcessEditCellProps` for non-editable columns when starting a row update (#17732) @arminmeh

Reviewer 2: ```suggestion
- [DataGrid] Ignore `preProcessEditCellProps` for non-editable columns when starting a row update (#17732) @arminmeh
- [DataGrid] Avoid applying row selection propagation on filtered rows (#17739) @MBilalShafi
```

#### `@mui/x-data-grid-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.3.0`.

#### `@mui/x-data-grid-premium@8.3.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.3.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@8.3.0`

- [DateTimePicker] Fix focus behavior on desktop variant (#17719) @LukasTy
- [pickers] Avoid `DigitalClock` stealing focus from a Picker open button on close (#17686) @LukasTy

#### `@mui/x-date-pickers-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.3.0`, plus:

- [DateRangePicker] Fix to reset range position after closing mobile Picker (#17631) @LukasTy

### Charts

- The `<FunnelChart />` series now accepts a `variant='outlined'` prop for a simpler style.

  <img width="398" alt="Screenshot 2025-05-06 at 20 36 12" src="https://github.com/user-attachments/assets/00fef14f-9026-421e-a4b6-7e081adce1e8" />

> - Add a `borderRadius` property to `<FunnelChart />`. All funnels have a `8px` `borderRadius` by default.

Reviewer 3: Do we need the backticks around 8px?

```suggestion
- Add a `borderRadius` property to `<FunnelChart />`. All funnels have a 8px `borderRadius` by default.
```

Reviewer 4: I prefer with the ticks, as that is technically code

Reviewer 3: > that is technically code

Not sure it is, it's just a value with its unit. Not a blocker, though

Author: ```suggestion
- Add a `borderRadius` property to `<FunnelChart />`. All funnels have `8px` as a default value.
```

Better maybe?

Reviewer 3: Sounds good to me 👍

  <img width="386" alt="Screenshot 2025-05-06 at 14 00 20" src="https://github.com/user-attachments/assets/4f4cc0e7-01ce-4ed6-a0e1-a387f78def23" />

- Add a `pyramid` curve to `FunnelChart`, this allows us to create a pyramid shaped funnel.

  <img width="344" alt="Screenshot 2025-05-06 at 14 32 59" src="https://github.com/user-attachments/assets/0b2896e0-0478-4766-bb1b-258a4977a751" />

- Add a `step-pyramid` curve to `FunnelChart`, which creates a stepped-pyramid like shape.
 
  <img width="344" alt="Screenshot 2025-05-06 at 14 33 03" src="https://github.com/user-attachments/assets/894f0ab3-7898-40fe-b0df-560feea4085a" />

#### `@mui/x-charts@8.2.1`

- [charts] Add charts toolbar with zoom options (#17615) @bernardobelchior

> - [charts] Add population pyramid demo (#17652) @bernardobelchior

Reviewer: This is a docs change. 🙈 
```suggestion
```

- [charts] Add zoom slider (#17496) @bernardobelchior
- [charts] Cleanup compiler warnings (#17360) @alexfauquette

> - [charts] Fix PieArcLabel not taking arcLabelRadius into account (#17655) @bernardobelchior

Reviewer: ```suggestion
- [charts] Fix `<PieArcLabel />` not taking `arcLabelRadius` into account (#17655) @bernardobelchior
```

- [charts] Fix spark line not having clip path (#17501) @bernardobelchior
- [charts] Fix type issue with ESM (#17624) @alexfauquette

> - [charts] Improve MarkElement perf (#17546) @bernardobelchior

Reviewer 3: ```suggestion
- [charts] Improve MarkElement performance (#17546) @bernardobelchior
```

> - [charts] Improve MarkElement perf (#17546) @bernardobelchior

Reviewer: ```suggestion
- [charts] Improve `<MarkElement />` perf (#17546) @bernardobelchior
```

> - [charts] Rename materialSlots internal constant (#17710) @bernardobelchior

Reviewer: ```suggestion
- [charts] Rename `materialSlots` internal constant (#17710) @bernardobelchior
```

- [charts] Update zoom slider design (#17682) @bernardobelchior

> - [charts] fix zoom being documented as available for heatmap (#17657) @bernardobelchior

Reviewer 3: ```suggestion
- [charts] Fix zoom being documented as available for heatmap (#17657) @bernardobelchior 
```

> - [charts] fix zoom being documented as available for heatmap (#17657) @bernardobelchior

Reviewer: ```suggestion
- [charts] Fix zoom being documented as available for heatmap (#17657) @bernardobelchior
```

#### `@mui/x-charts-pro@8.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.2.1`, plus:

- [charts-pro] Add `pyramid` curve to `<FunnelChart />` (#17665) @JCQuintas
- [charts-pro] Add `variant='outlined'` to `<FunnelChart />` series (#17661) @JCQuintas

> - [charts-pro] Add a `borderRadius` property to `<FunnelChart />` (#17660) @JCQuintas

Reviewer 4: yeah 👍

> - [charts-pro] Add a `borderRadius` property to `<FunnelChart />` (#17660) @JCQuintas

Reviewer: I very much agree on it and would also note that adding at least a small highlight wouldn't hurt. 🙈 👌

> - [charts-pro] Add `pyramid` curve to `FunnelChart` (#17665) @JCQuintas
> - [charts-pro] Add `variant='outlined'` to `FunnelChart` series (#17661) @JCQuintas
> - [charts-pro] Add a `borderRadius` property to `FunnelChart` (#17660) @JCQuintas

Reviewer: ```suggestion
- [charts-pro] Add `pyramid` curve to `<FunnelChart />` (#17665) @JCQuintas
- [charts-pro] Add `variant='outlined'` to `<FunnelChart />` series (#17661) @JCQuintas
- [charts-pro] Add a `borderRadius` property to `<FunnelChart />` (#17660) @JCQuintas
```

### Tree View

#### `@mui/x-tree-view@8.3.0`

> Internal changes.

Reviewer 5: Not sure why the changes were not created here 🤔 
```suggestion
- [TreeView] Fix keyboard navigation error (#17685) @rita-codes 
```

Internal changes.

#### `@mui/x-tree-view-pro@8.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@8.2.1`.

### Docs

> - [docs][charts] Fix randomised argos test (#17658) @JCQuintas

Reviewer 3: @JCQuintas does this make sense to be here?

Reviewer 4: yeah, it is not a component/bug fix

> 

Reviewer 5: Same here 👀 
```suggestion
- [TreeView] Bug fix - Escape does not cancel Drag n Drop (#17735) @rita-codes 
```

### Core

- [core] Avoid `node` types in the built packages (#17533) @LukasTy

> - [code-infra] Add pkg.pr.new publishing (#17402) @Janpot

Reviewer: ```suggestion
- [code-infra] Add `pkg.pr.new` publishing (#17402) @Janpot
```

> - [charts] Fix randomised argos test (#17658) @JCQuintas

Reviewer 3: ```suggestion
- [charts] Fix randomised argos test (#17658) @JCQuintas
- [docs] Make preview messaging consistent in charts @bernardobelchior 

```

> - [docs] Make preview messaging consistent in charts @bernardobelchior 

Reviewer: ```suggestion
- [docs] Make preview messaging consistent in charts @bernardobelchior
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.3.0

_May 7, 2025_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

> - 🎨 Added new styling options and shapes for FunnelChart, including `variant`, `borderRadius`, `pyramid`, and `step-pyramid` curves.

Reviewer: ```suggestion
- 🎨 Added new styling options and shapes for `<FunnelChart />`, including `variant`, `borderRadius`, `pyramid`, and `step-pyramid` curves.
```

- [DataGrid] Fix lazy loading crash with `isRowSelectable` prop (#17629) @MBilalShafi
- [DataGrid] Fix: use CSS nonce (#17726) @romgrk
- [DataGrid] Ignore `preProcessEditCellProps` for non-editable columns when starting a row update (#17732) @arminmeh

#### `@mui/x-data-grid-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.3.0`.

#### `@mui/x-data-grid-premium@8.3.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.3.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@8.3.0`

- [DateTimePicker] Fix focus behavior on desktop variant (#17719) @LukasTy
- [pickers] Avoid `DigitalClock` stealing focus from a Picker open button on close (#17686) @LukasTy

#### `@mui/x-date-pickers-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.3.0`, plus:

- [DateRangePicker] Fix to reset range position after closing mobile Picker (#17631) @LukasTy

### Charts

> - The `FunnelChart` series now accepts a `variant='outlined'` prop for a simpler style.

Reviewer: ```suggestion
- The `<FunnelChart />` series now accepts a `variant='outlined'` prop for a simpler style.
```

  <img width="398" alt="Screenshot 2025-05-06 at 20 36 12" src="https://github.com/user-attachments/assets/00fef14f-9026-421e-a4b6-7e081adce1e8" />

> - Add a `borderRadius` property to `FunnelChart`. All funnels have a `8px` `borderRadius` by default.

Reviewer: ```suggestion
- Add a `borderRadius` property to `<FunnelChart />`. All funnels have a `8px` `borderRadius` by default.
```

  <img width="386" alt="Screenshot 2025-05-06 at 14 00 20" src="https://github.com/user-attachments/assets/4f4cc0e7-01ce-4ed6-a0e1-a387f78def23" />

> - Add a `pyramid` curve to `FunnelChart`, which allows creation of a pyramid-shaped funnel.

Reviewer: ```suggestion
- Add a `pyramid` curve to `<FunnelChart />`, which allows creation of a pyramid-shaped funnel.
```

  <img width="344" alt="Screenshot 2025-05-06 at 14 32 59" src="https://github.com/user-attachments/assets/0b2896e0-0478-4766-bb1b-258a4977a751" />

> - Add a `step-pyramid` curve to `FunnelChart`, which creates a stepped-pyramid like shape.

Reviewer: ```suggestion
- Add a `step-pyramid` curve to `<FunnelChart />`, which creates a stepped-pyramid like shape.
```

>  

Reviewer: ```suggestion

```

- [charts] Fix zoom being documented as available for heatmap (#17657) @bernardobelchior

#### `@mui/x-charts-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.3.0`, plus:

- [charts-pro] Add `pyramid` curve to `<FunnelChart />` (#17665) @JCQuintas
- [charts-pro] Add `variant='outlined'` to `<FunnelChart />` series (#17661) @JCQuintas
- [charts-pro] Add a `borderRadius` property to `<FunnelChart />` (#17660) @JCQuintas

### Tree View

#### `@mui/x-tree-view@8.3.0`

- [tree view] Bug fix - Escape does not cancel Drag n Drop (#17735) @rita-codes
- [tree view] Fix keyboard navigation error (#17685) @rita-codes
- [tree view] Continue cleaning the plugin system (#17386) @flaviendelangle

#### `@mui/x-tree-view-pro@8.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@8.3.0`.

### Docs

- [charts] Add population pyramid demo (#17652) @bernardobelchior
- [charts] Fix randomised argos test (#17658) @JCQuintas
- [docs] Make preview messaging consistent in charts @bernardobelchior 

> 

Reviewer: ```suggestion
```

- [DataGrid] Fix lazy loading crash with `isRowSelectable` prop (#17629) @MBilalShafi
- [DataGrid] Fix: use CSS nonce (#17726) @romgrk
- [DataGrid] Ignore `preProcessEditCellProps` for non-editable columns when starting a row update (#17732) @arminmeh

#### `@mui/x-data-grid-pro@8.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.2.1`.

#### `@mui/x-data-grid-premium@8.2.1` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.2.1`.

### Date and Time Pickers

#### `@mui/x-date-pickers@8.2.1`

- [DateTimePicker] Fix focus behavior on desktop variant (#17719) @LukasTy
- [pickers] Avoid `DigitalClock` stealing focus from a Picker open button on close (#17686) @LukasTy

#### `@mui/x-date-pickers-pro@8.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.2.1`, plus:

- [DateRangePicker] Fix to reset range position after closing mobile Picker (#17631) @LukasTy

### Charts

> From https://github.com/mui/mui-x/pull/17661

Reviewer 4: ```suggestion
```

- The `FunnelChart` series now accept a `variant='outlined'` prop for a simpler style.

  <img width="398" alt="Screenshot 2025-05-06 at 20 36 12" src="https://github.com/user-attachments/assets/00fef14f-9026-421e-a4b6-7e081adce1e8" />

> From https://github.com/mui/mui-x/pull/17660

Reviewer 4: ```suggestion
```

- Add a `borderRadius` property to `FunnelChart`. All funnels have a `8px` `borderRadius` by default

  <img width="386" alt="Screenshot 2025-05-06 at 14 00 20" src="https://github.com/user-attachments/assets/4f4cc0e7-01ce-4ed6-a0e1-a387f78def23" />

> From https://github.com/mui/mui-x/pull/17665

Reviewer 6: ```suggestion
```

- Add a `pyramid` curve to `FunnelChart`, this allows us to create a pyramid shaped funnel.
  <img width="344" alt="Screenshot 2025-05-06 at 14 32 59" src="https://github.com/user-attachments/assets/0b2896e0-0478-4766-bb1b-258a4977a751" />
- Add a `step-pyramid` curve to `FunnelChart`, which creates a stepped-pyramid like shape.
  <img width="344" alt="Screenshot 2025-05-06 at 14 33 03" src="https://github.com/user-attachments/assets/894f0ab3-7898-40fe-b0df-560feea4085a" />

> From https://github.com/mui/mui-x/pull/17501

Reviewer 4: ```suggestion
```
